### DISCUSSION TITLE
Allow passing k/v pairs to a custom J2 template

### DISCUSSION THREAD

**Discussion by vachillo** [griptape-ai/griptape#1530] [Ideas]
- [x] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

**Is your feature request related to a problem? Please describe.**
I have custom J2 templates that i would like to use, but i have no way to pass custom k/v pairs without overriding the entire class

**Describe the solution you'd like**
I want to pass a dict to a Tool/Engine that use J2 templates that i can use to pass in k/v pairs to render to my custom template

**Describe alternatives you've considered**

**Additional context**

**Comment by collindutter** [comment]
This should be implemented through the following:
1. Add `task: self` to `BaseTask.full_context`. 
2. Add `context: self.full_context` as arguments to all Task uses of `J2.render`.

**Comment by collindutter** [comment]
@vachillo I know this is quite old but can you expand on the ask here? Can you provide a code snippet that shows what you're trying to do?

**Reply by vachillo** [reply]
if i have some kv pairs that i want to pass into the system prompt, i need to override the prompt to support that. 

> This should be implemented through the following:
> 
> Add task: self to BaseTask.full_context.
> Add context: self.full_context as arguments to all Task uses of J2.render.

It looks like this is implemented, I'm not 100% sure if its exhaustive though. There is still a question of user-friendly-ness when trying to use the J2 class and overriding the different template generator classes.

**Reply by collindutter** [reply]
I don't think [this](https://github.com/orgs/griptape-ai/discussions/1530#discussioncomment-11766129) ever got implemented, certainly not universally.

Why do you need to override any template generator classes? Can you provide an example of what you're trying to do?
